src/clr-angular/data/datagrid/providers/column-toggle-buttons.service.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, TemplateRef} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+export type ColumnToggleButtons = "ok"|"selectAll";
+
+@Injectable()
+export class ColumnToggleButtonsService {
+    buttons: TemplateRef<any> = null;
+    selectAllDisabled: boolean = false;
+
+    private _okButtonClicked = new Subject<any>();
+    public get okButtonClicked(): Observable<any> {
```

Comment:
I can't find where these two Observables are used. Do the buttons actually perform their respective actions? Based on the code in the PR, I don't see how that would be possible.

Reply:
Botched the original PR, pushing the right code which _uses_ the observables in the trigger component to call the corresponding methods.